marshal/consumer.go

```diff
@@ -145,7 +145,15 @@ func (m *Marshaler) NewConsumer(topicNames []string, options ConsumerOptions) (*
 					}
 				}
 			}
+
+			// this check needs to be after iterating all partitions in a topic
+			if len(c.claimedTopics) >= options.MaximumClaims {
```

Comment:
This was implied by claimedTopics as it's only updated for topic-claims. I can add the extra check if you need.